### ISSUE TITLE
ios11

### ISSUE THREAD

**Issue by jetmcquack** [nfarina/homebridge#1350]
I know it's definitely too early to talk about ios11 but i just wanted to report my experience so far.
I have noticed that any first accessory in config.json show up as "not Supported" and break the home configuration for every device, even those not on iOS 11.

**Comment by d0n13** [comment]
I have added a new accessory this eve and it seems to be behaving ok under iOS 11. No message in during startup like you are getting.

**Comment by jetmcquack** [comment]
Just for records this happens only on an iPad used as home hub

**Comment by fedor1210** [comment]
I haven't experienced this. Happily. Try to delete the "accessories" folder, may be this will help (warning: it will broke some plugins, for example homebridge-mqtt. You will need to add the accessories to it again).

**Comment by wild-a** [comment]
I’m running iOS 11 on my iPhone and iPad. The issue I ran into is that I’m unable to add homebridge using the Home app and the Eve app. I was able to add it using an iPhone that’s on iOS 10, and then all the accessories were working perfect.

**Comment by treeiler** [comment]
@wild-a there is a manual code button at the bottom of the pairing screen in the home app...worked for me.

**Comment by EdwinJS** [comment]
I have had the issue too, not sure sure of the cause but i have homebridge-cbus and had a light named "Entrance Chandelier" it was not supported when first adding the homebridge, i then stopped homebridge, renamed the light to "Entrance Chandelier 2", started homebridge, it then loaded and supported the light. Next I again stopped and renamed the light name in my config.json correctly and it remained supported and working... Not sure why it does that but it's fixed now

Hope that helped..

**Comment by wild-a** [comment]
@treeiler thanks! I missed that. I was able to add homebridge by manually entering the code

**Comment by KeyloggedUser** [comment]
Hey Everyone,

Just poppin in to say I too have the same issue. But I looked at what apple had to say about its new features to HomeKit and they indeed changed/improved it. Could be the connection either uses a different authentication method and/or the ports are different. iOS11 was never able to find my pi running homebridge.

Hope this is a simple fix for yuh @nfarina . Would be nice to have both awesome groundbreaking features (Homebridge and iOS11) together. Thanks for what you've put out so far! Great Work!
Have a good one!
~KLU

**Comment by bdruth** [comment]
For what it's worth, I can't get homebridge working "from scratch" on iOS 11. If I clear ``persist`` and ``accessories``, delete my home on the device, and try to setup from scratch - I go through the pairing process which is successful, but all devices report "No Response". Doing the same thing on my wife's iOS 10 device works as expected.

Not sure what the deal is, but I think we need to seriously look at what's changed in iOS 11 and start figuring out how to address. I'm happy to help, just point me in a direction :) I know my way around wireshark, node, and debuggers, but I'm not sure where it makes sense to get the needed info.

**Comment by pdlove** [comment]
I brought mine up from scratch without any problems in IOS 11 Beta 1. What plugins do you use and how many accessories are on the bridge? If one of the plugins is taking too long or there are 50+ devices on a single HomeKit instance, it has been known to cause issues.

**Comment by DanBennettUK** [comment]
It seems to have started failing in iOS 11 Beta 3. I'm also getting "No Response" for all accessories (which isn't many).

**Comment by acorkin** [comment]
B4 seems to have caused more issues on HomeKit in general. E.g. I am unable to set a specific time for automation on my standard HomeKit accessories, sensor trigger automation doesn't work also Homebridge isn't usable if you're away (using Apple TV (OS11) as a hub). I'm Assuming these issues are caused by tvOS 11. Hopefully, we will see some improvements in B5.

**Comment by DanBennettUK** [comment]
Yeah I find very rarely the connection will appear during the B3 stage but I've had no luck thus far. It's just... dead. And if I'm out of the network then there is just no chance (also using Apple TV)

**Comment by macjeff** [comment]
I am running beta 4 on iphone with no issues but I have not upgrade the Apple TV os.    So probably an issue there.  

Jeff

**Comment by chazzzle** [comment]
no response showing in homekit here as well, available for testing if needed

**Comment by macjeff** [comment]
I just confirmed this and have a workaround for now.  

The appletv is doing it.  Everything was working for me until I turned on the Apple TV.   (I did not realize it was unplugged when I said it was working already).  

First you have to erase all bridges in your home app and stop them from running in your server. 

Then turn HomeKit hub off in Apple TV.  

Then in homebridge data folder delete the persists and accessories folders. 

Then open homebridge again on server.  Then add bridges back to device.  

Now you are good but don’t turn Apple TV hub on.   As soon as you turn it on within about 15 minor less your devices will either disappear or say no response.   

I am going to send bug to Apple on this too.  

Jeff

**Comment by FradSer** [comment]
@macjeff thank you for this.

**Comment by DanBennettUK** [comment]
Interestingly I can't even find the homebridge hub at all after removing it. No option on the Apple TV either, seems like Apple TV knows I've removed it.

This is odd.

**Comment by bdruth** [comment]
I turned the Apple TV on and started from scratch (leaving the ATV off) and I was able to get things working again. On iOS 11 B4. Hopefully we'll be able to use ATV again soon.

**Comment by macjeff** [comment]
Good news.

I took the time today and updated BOTH my apple tv’s to TV OS 11 b4.

It seems to be working!!!

**WARNING**  you must have your device registered either under the developer or public beta to use the beta.   I would like a few beta testers that are willing to test my theory and future theory and provide feedback.  I have direct line to some apple engineers so I can get things done.

Please let me know if you are interested and I will tell you the information that I need.

But if you are already registered or you can get your own public beta or developer account if you update your apple tv (And then you may have to reset up home app one last time) then you should be fine but I will let you know if anything happens but its been longer than ever and I actually have TWO apple ID’s signed in now.

(PS you will love the new home screen sync feature that will make all the folders and apps identical throughout your house on multiple apple tv’)

Jeff

**Comment by bdruth** [comment]
@macjeff I've been using an ATV3 (with it's respective limitations) - does it seem like iOS 11 is just not going to support that anymore? I've been considering getting an ATV4, but haven't pulled the trigger yet.

**Comment by DanBennettUK** [comment]
Thing is Jeff I already had tvOS on beta 4. 

Removed everything. Can’t can find the accessories now at all. Nothing in debit output either. It’s like iOS isn’t even trying :-/


Regards,

Dan Bennett

**Comment by macjeff** [comment]
No.   Everything in testing is all for Gen 4

Jeff

**Comment by macjeff** [comment]
Are you sure you removed the accessories folder?   If you still have an issue it could be a bad plugin.   Try only running one at a time.

Jeff

**Comment by ekgerke** [comment]
Sorted out the same issue with my aTV4 and homebridge. All I had to do was to turn homekit off and then on again under Settings > Accounts > iCloud > Homekit. Now all accessories work again both on home wifi and LTE

**Comment by macjeff** [comment]
i guess I stumbled upon this by deleting my entire home account and all homebridge data and started from scratch but the problem kept coming back until I went to TV os 11 beta 4

Jeff

**Comment by erfinb** [comment]
I am now having issues with everything just reporting as "updating" that is connected to my homebridge.... anyone else? Newest betas of iOS 11 on all devices (Apple TV 4, iPhone, and iPad).

**Comment by leonmf** [comment]
I’m having the same problem as everyone else. My bit to add to the equation is this. 

* my official HomeKit devices still work just fine. 
* both homebridge and the HomeKit integration in Z-Way fail in exactly the same way. 
* I have one z-wave dinner switch that keeps disappearing from HomeBridge and Z-Way HomeKit. I think it does this when I assign it to a room. I haven’t had much time to test. 

My setup:
My iPhone iOS 11 b4
Atv4, iPhone, 2 iPads all on iOS 10.whatever the latest is. 

At the moment I’m willing to live with it broken on my phone but don’t want to upgrade anything else at the moment because those, at least, all (mostly) work.

**Comment by devbobo** [comment]
I've experienced some issues with 2 of my platforms. The symptoms appeared to be the same between the 2 platforms.

On both platforms, it appears that the 5th accessory defined by the plugin would become blocked.

**Comment by DanBennettUK** [comment]
> z-wave dinner switch

I need me one of those.

An update on my part... after removing everything from HomeKit, resetting Persist and Accessories directories and completely hard rebooting the aTV (tvOS 11 B4). I tried to connect and the HomeBridge wasn't found.

Left it a few hours... tried again and it worked! My Flic switch appears as not supported though, but I at least have control back again...

I think the aTV caches information like this, even if you remove it completely. I've had this issue before where things consistently fail to work until you just give up for a few hours. There doesn't seem to be a way to get the aTV to clear it's cache...

I wonder what is causing it, though. Is it a bug that's just disconnected HomeKit devices in total or is it something affecting HomeBridge itself? Has anyone tried the raw NodeJS HAP stuff to see if that also fails under the same conditions?

**Comment by ekgerke** [comment]
I'm using homebridge-edomoticz plugin and have had a few accessories show up as "unsupported" aswell. The trick to fix them has been creating a new accessory with a different name and have it replace the original one.  There seems to be a caching issue as although I renamed a device in the domoticz side of the system, the home app still kept on showing the original name for the device. The only way I got the name to be right was when I completely reset the homebridge username in config.json and set it up all over again. And again.. I also had to make sure both of my homekit hubs (aTV and iPad) were toggled.

**Comment by SPINACHCEO** [comment]
@DanBennettUK I have the same problem re-add bridge again.
 I found that you need to change both the "username" and "pin" in homebridge config file to make HomeKit able to discover it again.

Well, anyway it works for a while after re-add them. After a couple day all my accessories back to "no response" again. One more "re-add" works, but I am not sure if it will break again then.

@ekgerke FYI, as @macjeff said, remove all accessories caches should work.

**Comment by SPINACHCEO** [comment]
Anyway is the a bug in beta 4 or they are really just trying to break homebridge ?

**Comment by andre-dw** [comment]
Is this fixed with the newest beta?

**Comment by macjeff** [comment]
No.  Don’t think it will be fixed by Apple. 

Jeff

**Comment by leonmf** [comment]
I have the same problems in either homebridge or the HomeKit implementation built into z-way. 

So, not necessarily specifically targeted but I doubt Apple is staying awake worrying about us. 

If anyone made a z-wave to HomeKit bridge, I’d be telling them to shut up and take my money. 

Leon Grossman
Leongrossman@gmail.com

**Comment by pdlove** [comment]
To update something on this... If all you're experiencing is a single device showing as "Not Supported" then that is a bug with HomeKit in IOS 11. If this does not get fixed, we'll figure out a work around. As far as the devices stuck at "Updating" or "Not Responding"... I haven't been able to reproduce that reliably enough to try and figure it out, although I would be surprised if it isn't related.

**Comment by macjeff** [comment]
the .25 version still had the NOT SUPPORTED until I deleted the devices and set them up from scratch.

Now no more issues!!!

Jeff

**Comment by pdlove** [comment]
Awesome. We have one more update tomorrow to fix a few issues with various Plugins. You shouldn’t have to re setup with that update.

**Comment by macjeff** [comment]
I have 3 instances running.   After several hours all three become unresponsive on all devices.   I am not home now but hopefully terminating processes and starting it up again will fix it.   If not I will have to reset it and lose all my room assignments again.  

Jeff

**Comment by DanBennettUK** [comment]
Have the same issue today. All devices are no longer responding until after
a restart of the service. Will check again in a few hours (after sleep).
And this is after resetting fully and re-adding everything.

**Comment by macjeff** [comment]
Did you try .26 just released?

**Comment by pdlove** [comment]
After this update, if anyone gets the not responding.... Let's talk about it.  One thing I'll need from a not responding system is the accessories dump which you can only get running in insecure mode.

**Comment by fedor1210** [comment]
I had iOS 11 beta 1 in June and reversed to iOS 10... but I needed to reset the "user-id" (that string like MAC) and delete the "present" folder to stop HomeKit revert any changes such as changing room/name/icon of the accessory or even favorites after 5 seconds... I still have this problem as it returns after about 1 hour. I believe it is not problem with HomeBridge, but with iOS (of course :) )

**Comment by justinanastos** [comment]
@pdlove I updated `homebridge` and all functionality appears to be restored using the lastest public iOS b6.

**Comment by alfred011** [comment]
I can also confirm that upgrading homebridge (Sudo npm -g upgrade homebridge) made all unresponsive devices work again. This is also on iOS 11 Public Beta 5. 
Previous to the upgrade only iPhone 7 with PB5 worked, iPad as home hub PB5 showed unresponsive. 
Did not have to remove homebridge from Apple Home.

**Comment by macjeff** [comment]
Beta 6 by itself did not work for me

And .25 version kept loosing connection. 

But 24 hours up on .26 and still working.   And I am remote and all devices giving status.  

Jeff

**Comment by DanBennettUK** [comment]
.26 has now failed for me with all devices showing No Response :(

**Comment by macjeff** [comment]
Just checked.   Mine still going strong.   But I did start over from .25 to .26

Jeff

**Comment by pdlove** [comment]
@danbennettuk which Plugins?

**Comment by DanBennettUK** [comment]
@pdlove Magichome plugin, to control the actual lights. And a WOL plugin (accessory "computer").

They all work after a restart (though sometimes it takes time to decide to work). But then dies again not long after...

**Comment by pdlove** [comment]
From what I'm reading... if any command fails to complete within 10 seconds, the security has to be re-negotiated. This isn't currently supported by HAP-NodeJS and could cause the Not Responding messages until Homebridge is restarted. Is it possible that any of the get or set processes on your plugins are taking longer than 10 seconds?

**Comment by DanBennettUK** [comment]
Sorry - incredible delay.

Usually, there isn't really much of a delay at all. Until iOS 11, I've had no issues. With iOS 11, all devices sometimes don't respond or never respond (the WOL "switch" also stops responding at the same time).

I've attempted to reinstall everything again since the latest beta. Currently not even able to see the homebridge device, let alone add the devices... so now working out how to solve that. Before this, I could add devices but they were appearing as not available (forget actual wording) instead of the red "No Response" message!

**Comment by jpablomunoz** [comment]
Solved for me... iOS 11.0.1 (15A402) work correctly.. 
Delete Bridge on HomeKit app on iPhone 
then 
Run an edit  MAC Address on config.json  and  config-sample.json
 sudo nano /home/pi/.homebridge/config.json
save
 sudo nano  /usr/local/lib/node_modules/homebridge/config-sample.json
save
sudo /etc/init.d/homebridge restart
sudo /etc/init.d/homebridge stop
sudo /etc/init.d/homebridge status
killall homebridge
sudo /etc/init.d/homebridge status
sudo /etc/init.d/homebridge start
sudo /etc/init.d/homebridge status

and then 
search on home kit app your bridge, then add  enroll  number manually , finally when found bridge  config normally.-

success..

**Comment by B1tMaster** [comment]
Problem solved: https://github.com/B1tMaster/haas-config/issues/54
This is the best way to solve it as it is pretty much guaranteed to work, since you would be installing the homebridge plugin via Docker and your environment, libraries , settings etc will be same regardless of what OS you run.

**Comment by LeoCal** [comment]
Folks, in the end I managed to get everything working again. No need to run homebridge in Docker or to install Hass.io. What you need to do is a cleanup of your PI.
I basically removed all the packages (nodejs, npm, node-server, nodered, hap-nodejs, homebridge) and installed everything from scratch following the installing procedures at:
- https://github.com/KhaosT/HAP-NodeJS
- https://github.com/nfarina/homebridge
Special thanks to the support from @KhaosT and @nfarina while I was trying to troubleshoot their code (in the end their code was just fine :-)).

**Comment by icalphil** [comment]
My experience with Home Bridge is positive. I did the homebridge using MAC.

Couple Note here for first timer:
1. if you have NPM permission issue while installing homebridge and homebridge-smartthing, try follow https://docs.npmjs.com/getting-started/fixing-npm-permissions , once it is done, re-install again.
2. This video does help on setting everything up: https://www.youtube.com/watch?v=9yQklSq3_Mo
3. Topic here help to understand what needs to be setup in general https://community.smartthings.com/t/hello-home-homekit-and-siri-control-via-homebridge/16701
4. Lots of people asking where to put "config.json", here is the path, "/Users/your user name/.homebridge" (Please make sure you have installed the homebridge and homebridge-smartthing before go search for this folder, otherwise, you will not find it.
4. in case you got the message from your HomeKit said that you need to "reset the homekit" in order to add the accessory again, what you need is go to config.json file, change the username, you can do so by modify it for example "CD:22:3D:E3:CE:33" to "CD:22:3D:E3:CE:40" and change your pin, example from "035-45-855" to ""035-45-860" and restart homebridge again. You should be able to add those accessory again. hope that helps!
